polars/polars-io/src/csv/utils.rs

```diff
@@ -112,18 +111,13 @@ fn infer_field_schema(string: &str, try_parse_dates: bool) -> DataType {
             {
                 match date_infer::infer_pattern_single(&string[1..string.len() - 1]) {
                     Some(pattern_with_offset) => match pattern_with_offset {
-                        PatternWithOffset {
-                            pattern: Pattern::DatetimeYMD | Pattern::DatetimeDMY,
-                            offset: _,
-                        } => DataType::Datetime(TimeUnit::Microseconds, None),
```

Comment:
if we're converting to UTC, we no longer need `PatternWithOffset`, `Pattern` is enough